### ISSUE TITLE
Support Mac groups vCards 

### ISSUE THREAD

**Issue by nalt** [nextcloud/contacts#213]
### Steps to reproduce
1. Setup CardDAV Sync in MacOS
2. Create a group in MacOS Contact app

or

1. Create a group in Nextcloud

### Expected behaviour
Groups created in Contacts app should appear as groups in Nextcloud.
Groups created in Nextcloud should appear as groups in Contacts app.

### Actual behaviour
A group created in Contacts app appears as a normal contact in Nextcloud.
Nextcloud groups are not represented in Contacts app.

### Server configuration

**Operating system**: Nextcloud official Docker

**Web server:** Nginx Proxy

**Database:** MariaDB

**Nextcloud version:** 11.0.3

**Contacts version:** 1.5.3

**Updated from an older Nextcloud or fresh install:** Updated, but contacts cleared

### Client configuration
**Operating system:** MacOS 10.12.4

**CardDAV-clients:** MacOS Contacts app

**Comment by EddyLB** [comment]
## Same issue with Nextcloud 12

### Server configuration
**Operating system**: Debian GNU/Linux 8.8 (jessie)  x86_64

**Web server:** Apache/2.4.10 (Debian) (apache2handler)

**Database:** mysql 5.5.55

**PHP version:** 5.6.30-0+deb8u1
Modules loaded: Core, date, ereg, libxml, openssl, pcre, zlib, bcmath, bz2, calendar, ctype, dba, dom, hash, fileinfo, filter, ftp, gettext, SPL, iconv, mbstring, session, posix, Reflection, standard, shmop, SimpleXML, soap, sockets, Phar, exif, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlwriter, zip, apache2handler, PDO, apcu, curl, gd, imagick, intl, json, mcrypt, memcache, mysql, mysqli, pdo_mysql, pdo_sqlite, pspell, readline, recode, sqlite3, tidy, xmlrpc, xsl, mhash, apc, Zend OPcache

**Nextcloud version:** 12.0.0 - 12.0.0.29

**Are you using external storage, if yes which one:** files_external is disabled

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

### Client configuration
**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/603.3.1 (KHTML, like Gecko) Version/10.1.2 Safari/603.3.1

**Operating system:**
- macOS Sierra version 10.12.6 Beta (16G8c) and Contacts version 10.0 (1756.20)
- iOS 10.3.3 (14G5037b)

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from last release of NextCloud 11

**List of activated apps:**
<details>
	<summary>App list</summary>
```
Enabled:
 - activity: 2.5.2
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.3
 - comments: 1.2.0
 - contacts: 1.5.3
 - dav: 1.3.0
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_pdfviewer: 1.1.1
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - issuetemplate: 0.2.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0

Disabled:
 - admin_audit
 - encryption
 - files_external
 - gallery
 - user_external
 - user_ldap
```
</details>

**Comment by skjnldsv** [comment]
Could you create a new contact on your Mac and sync it to nextcloud, then export it from nextcloud and post the result here please.

**Comment by irgendwie** [comment]
Afaik MacOS (and/or Apple in general) stores groups as vcards and not as property in a vcard, like we do it in the contacts app. When configuring davDroid (for Android) you can choose between those two options, I'm not sure if this is also possible for apple products.

**Comment by skjnldsv** [comment]
If we want to implement this, this won't be a priority then :/

**Comment by EddyLB** [comment]
Hi @skjnldsv,
I've created a group "macOS" and a contact in that group. As expected, mac OS Contacts creates 2 cards:

- one declaring the group with `X-ADDRESSBOOKSERVER-KIND:group`
and the ID `X-ADDRESSBOOKSERVER-MEMBER:urn:uuid:8a3da972-8df0-4e5a-a572-bdd4621abc39`. Apple uses `X-ADDRESSBOOKSERVER-KIND` instead of `KIND`, which is to my understanding acceptable according to RFC 6350.

- one declaring the contact and attaching the card to the group using the ID with `UID:8a3da972-8df0-4e5a-a572-bdd4621abc39`

After syncing with Nextcloud, I get two non-grouped new cards (attached). It would really be great that the Nextcloud Contacts app accepts `X-ADDRESSBOOKSERVER-KIND` as well as `KIND` for identifying vCard representing a group of persons or entities.

[Group macOS.vcf.txt](https://github.com/nextcloud/contacts/files/1179015/Group.macOS.vcf.txt)
[Contact In macOS Group.vcf.txt](https://github.com/nextcloud/contacts/files/1179030/Contact.In.macOS.Group.vcf.txt)

**Comment by skjnldsv** [comment]
Isn't there an option to change this behavior on Mac?

**Comment by EddyLB** [comment]
On macOS 10.12, the user can choose between the vCard formats 2.1 and 3.0 (and the vCard encoding format for 2.1). He/she can also enable/disable the export of notes and photos in vCards and the use of a private me card.

As shown for example [here](https://en.wikipedia.org/wiki/VCard), the property `KIND` is not supported in 3.0 (only in 4.0).

**Comment by nalt** [comment]
MacOS does not have option to change how the vCard are synced. Even if there was a hack via some config file, this would not be user-friendly at all. Their solution corresponds to the KIND/MEMBER fields of vCard 4.0, without breaking the older standards. So the logic is the same as in the 4.0 standard.

How does Nextcloud store groups - as a name in the Categories field? What would happen, if a vCard 4.0 using the standardized KIND/MEMBER fields is sent?

I would assume the version option exposed to the user only affects the export functionality.

**Comment by aluxnimm** [comment]
We support iCloud contact groups with X-ADDRESSBOOKSERVER-KIND:group also now for mapping Outlook Distribution Lists to contact groups in [OutlookCalDavSynchronizer ](https://github.com/aluxnimm/outlookcaldavsynchronizer/)

So would be really useful if the contact app could show the members of the group for a better Nextcloud and Outlook integration! Now the group is only visible as a normal contact with the name of the group and no members but atleast can be synced back to (another) Outlook or macOS instance.

**Comment by skjnldsv** [comment]
Let's go!
Adding this to the 2.1 milestone! We'll try implementing that!

**Comment by nalt** [comment]
Great!
Do you have a plan already how to handle this vCard "dialect" and potential other ones?

**Comment by sushidave** [comment]
FYI - The same issue exists for ownCloud: [#475](https://github.com/owncloud/contacts/issues/475).

**Comment by Woi** [comment]
Maybe nice to know:
DAVdroid has a [nice comparison table](https://www.davdroid.com/faq/cant-manage-groups-on-device/) of both methods.
They also document the [limited group support in nextcloud app](https://www.davdroid.com/tested-with/nextcloud/).

**Comment by chrsch** [comment]
@skjnldsv Did you have any success with this? Is it still ongoing or stale?

**Comment by skjnldsv** [comment]
It's still ongoing.
Can some of you try a vcard 4 group with the real KIND and not X-ADDRESSBOOK-KIND property on mac to see if it works?

**Comment by dphi** [comment]
Hi @skjnldsv, is does not work for me with the following adjustments:

Original: 
> VERSION:3.0
PRODID:-//Apple Inc.//AddressBook 11.0//EN
N:Test-Group;;;;
FN:Test-Group
X-ADDRESSBOOKSERVER-KIND:group
X-ADDRESSBOOKSERVER-MEMBER:urn:uuid:909c0bfb-9a8b-4514-b7d4-0b41ba7b11d2
REV:2018-04-07T20:45:06Z
UID:b450093e-e90c-45f6-8bd4-3b91facd123
END:VCARD

Adjustment:
>VERSION:3.0
PRODID:-//Apple Inc.//AddressBook 11.0//EN
N:Test-Group;;;;
FN:Test-Group
**KIND**:group
**MEMBER**:urn:uuid:909c0bfb-9a8b-4514-b7d4-0b41ba7b11d2
REV:2018-04-07T20:45:06Z
UID:b450093e-e90c-45f6-8bd4-3b91facd123
END:VCARD

**Comment by skjnldsv** [comment]
Strangely enough, KIND/MEMBER are supposed to be supported for v4+ only :)
I guess we'll try to add a support for both then.

**Comment by aluxnimm** [comment]
Yes a vcard with VERSION:3.0 is not valid with KIND/MEMBER, that's the reason for X-ADDRESSBOOKSERVER prefix. Only with VERSION:4.0 KIND/MEMBER are allowed, so the adjustment above is an invalid vcard. You really should support both, since there is no support for v4+ cards in many clients/servers.

**Comment by sushinger** [comment]
Hi, since I'd love to see this feature implemented, I'd like to help by testing with NC (latest version that I can install via Softaculous on my cPanel) in conjunction with two mac clients:
Contacts.app on Mac OSX 10.11 (El Capitan) and macOS 11.13 (High Sierra)

What is the recommended testing procedure? I assume to connect NC CardDAV with both clients and then add a contact on each client and on NC, then create a group on each client and on NC, then sync them all and post the result here?

Whatever makes sense to contribute, please let me know. I will be gladly helping here as a tester with the above mentioned Mac-clients :-)

**Comment by skjnldsv** [comment]
Hey @sushinger :)
We do not require testing yet because this feature is not implemented yet and will require a lot of changes on the server.

Nonetheless, we'd gladely need help when we'll start working on this ;)

**Comment by foux** [comment]
Whenever you need tester, I'll be available too ;)

**Comment by skjnldsv** [comment]
You folks are amazing! :D

**Comment by pagaille** [comment]
Just chiming in to tel you that Busycontacts also expect the server to support cards with X-ADDRESSBOOKSERVER-KIND:group...

**Comment by Aesculapius** [comment]
So as I understand it this feature is needed to allow regular clients like emClient to display groups/categories correctly? That's quite a missing feature at the moment... Is there any roadmap about when this could be implemented?

**Comment by skjnldsv** [comment]
Process:
- [ ] Migration path 
  - [ ] Warning to the admin
    Extensive process
  - [ ] Notify the user
    - If any davx was used lately, show documentation for it
- [ ] Support mac groups
- [ ] Add delete support #1175 
- [ ] Add rename support

**Comment by alesf** [comment]
Is it possible that something changed in iOS 13 (possible even on macOS Catalina - not tested) on how groups are handled?
All of the group vcards that have X-ADDRESSBOOKSERVER-KIND:group are shown as contacts and do not show in the groups.

**Comment by docbobo** [comment]
Here's what I am seeing:
- Groups created in Nextcloud do not appear on MacOS or iOS
- Groups created on MacOS or iOS do show up as regular contacts on Nextcloud

**Comment by microfx** [comment]
Any news on this? I have a lot of contacts not showing up which should show up in contacts.app on macOS Catalina.

**Comment by skjnldsv** [comment]
@microfx as you can see in the first post, no progress has been made. I have other priorities and no other dev started working on it :)

**Comment by microfx** [comment]
Thanks for letting me know. This is a pretty severe problem imho since it makes nextcloud not usable for a team and shared contact groups in an Apple ecosystem right now. So if there is a manual workaround it would be very nice to know. 

Or do I misunderstand this issue - could you shortly explain what groups vCards are? My problem is that not all contacts are shown on iOS which were shared from another user/group in nextcloud.

**Comment by pagaille** [comment]
That’s because there are two ways of creating groups. Basically there is the Apple way and the Google / Android way. Nextcloud (actually the underlying SabreDAV library) supports only the Android way. It’s a shame since if I’m not mistaken Apple follows the RFC much closer.

The only workaround I found for OSX  is buying Busycontacts <https://www.busymac.com/busycontacts/> which IMHO is pretty much worth the value. They also had problems supporting Nextcloud but issued a fix quickly after I asked them to look into this problem. NB : I’m not affiliate with them in any way 😊

**Comment by SuperSandro2000** [comment]
@pagaille Apple way: pay 50 bucks to fix a problem.

**Comment by pagaille** [comment]
@SuperSandro2000 : Google way : do things their own way and claiming that they use open standards. The same goes for their very personal IMAP protocol implementation with gmail BTW.

**Comment by microfx** [comment]
Android is not even supporting Nextcloud natively ...at least not the calendar. I don't get why Nextcloud is doing anything the "android way" if that protocol is more or less from apple (isn't it?).

That app doesn't look like I wanted to use it. And 50 Bucks for an app I don't like is too much. 

> Am 02.01.2020 um 14:19 schrieb Matthieu <notifications@github.com>:
> 
> ﻿
> That’s because there are two ways of creating groups. Basically there is the Apple way and the Google / Android way. Nextcloud (actually the underlying SabreDAV library) supports only the Android way. It’s a shame since if I’m not mistaken Apple follows the RFC much closer.
> 
> The only workaround I found for OSX is buying Busycontacts <https://www.busymac.com/busycontacts/> which IMHO is pretty much worth the value. They also had problems supporting Nextcloud but issued a fix quickly after I asked them to look into this problem. NB : I’m not affiliate with them in any way 😊 
> 
>

**Comment by olantrust** [comment]
This discussion seems a bit confusing. I don't think Android are not following the protocol (for vCard 3), Apple in our view are less than Android it appears, using X- properties.

In any case, vCard 4 fully supports both 
a) CATEGORIES (which are to be used as tags/labels and not to be used for group contacts) and 
b) KIND: GROUP (which creates an entity connecting the various entities).

SabreDav supports both when using vCard 4(?)
Apple and it's adapters appears to support both, when using vCard 4(?)
Android and it's adapters (such as also appears to support both, when using vCard 4(?)

If that is correct, this issue relates only the vCard 3.
Or to the vCard 4 where Nextcloud (or other software) incorrectly changes contacts to vCard 3.

Theoretically, an up-to-date system shouldn't suffer from this issue (unless they're using some software on Windows or Linux desktop, or an old Mac OS)?

That's not to belittle the issue, but just wondering whether there is another problem here - the unwanted conversion to vCard 3 for example here: https://github.com/nextcloud/server/issues/12878

**Comment by microfx** [comment]
It would be nice if my problem wasn't related to this issue. 

What I found out so far: 

If I create in contacts.app in macOS Catalina (latest) a group in nextcloud a new contact with that name gets created.

I cannot reconstruct which groups are being synced to my contacts.app (it's definitely a mixture somehow...) but syncing both ways works with those that are there. 

Also I guess I have somehow mixed up the words groups and addressbooks maybe. Still don't know how to use them properly tbh. But the protocol seems to not now this either.

Any idea what I could try? I have shared addressbooks in nextcloud - I think somewhere there lies the culprit.

**Comment by olantrust** [comment]
@microfx  do you have the text of a test contact vcf file - for the group and individual contacts?

Also, from what I can read, your issue seems to relate to contacts not showing in iOS, not in Nextcloud or MacOS, is that correct? Also what version of iOS are you using?

Thanks

**Comment by microfx** [comment]
in iOS 13 everything works fine - it's just macOS which is not working as expected (broken as in not showing all shared addressbooks / contacts I can see in iOS or of course in nextcloud).

iOS 13 and macOS 10.15

**Comment by plinss** [comment]
Just to be clear, using macOS Catalina (10.15) and iOS 13.3:

* You can create groups in the Contacts app on the Mac, there is no way to create groups on iOS (that I know of). These groups work and sync as expected on the Mac and iOS, however they show up in the Nextcloud Contacts app as a contact, not a group.

* If you create groups in the Nextcloud Contacts app, those groups do not show up on either the Mac Contacts app or iOS.

**Comment by microfx** [comment]
no. iOS is all fine. macOS is the problem.

**Comment by plinss** [comment]
> no. iOS is all fine. macOS is the problem.

I beg to differ, I just created a group in the Nextcloud Contacts app, and it does not show up on iOS.

**Comment by microfx** [comment]
are we talking about nextcloud user groups here? which then named shared addressbooks? because that is what I am using / having problems with (only macOS!).  

> Am 06.01.2020 um 00:27 schrieb Peter Linss <notifications@github.com>:
> 
> I beg to differ, I just created a group in the Nextcloud Contacts app, and it does not show up on iOS.
> 
>

**Comment by plinss** [comment]
> are we talking about nextcloud user groups here? which then named shared addressbooks? because that is what I am using / having problems with (only macOS!).

I am not. I am taking about contact groups created in the contacts apps, these are different than Nextcloud user groups and also different than having multiple address books. 

This issue is about contact groups, iOS and macOS store contact groups as a type of vCard, where the Nextcloud Contacts app does not recognize those vCards as groups and displays them as a contact. Furthermore the Nextcloud Contacts app stores group information in the contact's vCard, which neither macOS or iOS recognize.

It sounds like you are creating different address books entirely and sharing those among Nextcloud user groups, which is fine. When you create multiple address books in Nextcloud, iOS Contacts represents those visually the same way that it does contact groups, so this is where you're likely getting confused, but they are distinct concepts in the way the data is stored.

macOS Contacts app does not appear to recognize multiple address books, which is the problem you are facing.

Try creating a group in the macOS Contacts app and drag one of your contacts into that group, you'll see the issue.

**Comment by microfx** [comment]
a few posts above I described what happens when I create a group in macOS - the group will show up as a contact in nextcloud. 

Thanks for enunciating my issue.

**Comment by alesf** [comment]
This has already been discussed above (see https://github.com/nextcloud/contacts/issues/213#issuecomment-318271683)


I try to add necessary information to a group vCard so all clients can understand and group contacts.

Group vCard:
```
VERSION: 3.0
FN: Group Name
N: Group Name
UID: group-uuid-XXXX
X-ADDRESSBOOKSERVER-KIND: group
KIND: group
```

Contact vCards that belong to this group also have these fields:
```
X-ADDRESSBOOKSERVER-MEMBER: urn:uuid:group-uuid-XXXX
MEMBER: urn:uuid:group-uuid-XXXX
CATEGORIES: Group Name, ...
```

**Comment by skjnldsv** [comment]
same as https://github.com/nextcloud/contacts/issues/609

**Comment by nalt** [comment]
Please keep this issue open until the bug is fixed. #609 describes a feature, doesn't mention any details about the bug.

**Comment by Uffje** [comment]
On iOS 14, groups created in Nextcloud do not show up either. I am not using MacOS 11 so cannot test that. Like others said, contacts created on MacOS 10.15 show up as contact cards and dissapear as soon as I select them on Nextcloud.

**Comment by kittonian** [comment]
I've read through all of the above and I'm seeing different behavior. On Mac OS High Sierra, I can create a contact group and I see nothing in the web gui. However, I see that group across the desktop and iOS Apple clients on the LAN. Some of the desktops are High Sierra and others are the latest Catalina. All iOS devices are running the latest version.

If I create and/or delete a group, all devices update themselves, but nothing is populated in the groups. Except for the initial system where the contact group was created and various contacts were added to the group, all other devices show the group but the group is empty.

I have tried creating the group via the web gui, via other Mac OS desktops, etc. It's consistently the same behavior. The groups show and disappear properly but they are always empty (i.e. nothing is sync'ing inside the groups).

We are running Debian, PHP 7.4.8, and the latest Nextcloud (19.x).

**Comment by EddyLB** [comment]
This issue is opened for almost 3 years and I regret very much that it is still impossible to sync groups with Apple devices.

**Comment by brezlord** [comment]
It's 2023 and this important issues is still not resolved. When will this be fixed? I feel this issues is more important than AI in Nextcloud.

**Comment by Matteo85** [comment]
> Afaik MacOS (and/or Apple in general) stores groups as vcards and not as property in a vcard, like we do it in the contacts app. When configuring davDroid (for Android) you can choose between those two options, I'm not sure if this is also possible for apple products.

With regards to issue: Contact groups can't be deleted #1175

It sounds like Apple isn't following the vCard 3.x or 4.x standards.  Since Nextcloud is built to sustain those standards, waiting to fix an Apple mistake doesn't serve the community, especially since simple issues like being able to delete a user group after creating it (a simple CRUD fix) is being held back by an issue that may _never_ be resolved.  Apple has a tendency to do things the way they want and try to force compliance with their modus operandum.  Since it is Nextcloud's intention to follow the vCard standards:

_This is a complicated answer. We did not invent the way contacts works. We are following the official vCard format. This format include a predefined set of properties that other applications/devices supports too (Android, iOS, Windows, Gnome...). Adding a custom property would just make it compatible with Nextcloud and will not be understood by any other clients and make it complicated for us to ensure its sustainability over time. Long story short, we suggest you use the Notes field to add your custom data :)_

I would suggest moving forward with vCard standards UI and functionality issues and not wait for Apple.  You can't please everyone, and this is a glaring error (#1175)

**Comment by rokejulianlockhart** [comment]
@Matteo85, should we post that in the Apple Developer Forums then?

**Comment by Matteo85** [comment]
You can if you want.  I posted here because another thread (#1175), which is not part of the Mac development, referred to this error as having to be fixed before the thread I was viewing would be fixed.  In my opinion this delay is ridiculous since they are unrelated items (or should be...).  We should be able to delete user groups after creation and to have that basic, simple functionality delayed for a niche problem shouldn't be happening.

**Comment by nano13** [comment]
> You can if you want. I posted here because another thread (#1175), which is not part of the Mac development, referred to this error as having to be fixed before the thread I was viewing would be fixed. In my opinion this delay is ridiculous since they are unrelated items (or should be...). We should be able to delete user groups after creation, and this has been a known issue since 2019. To have basic CRUD functions delayed for a niche problem shouldn't be happening.

And again a year later, no solution in sight, and still blocking #1175 , which is an absolutely basic standard feature, still missing for at least half a decade now. I'm very sorry to say, but this is really just a huge shitshow ...

**Comment by microfx** [comment]
yeah ... haha ... ain't nobody using nextcloud for serious stuff anymore due to lacking features like this.

**Comment by nano13** [comment]
So I am actually an IT-Admin running a growing company offering typical services to a bunch of customers. I would love to have a real alternative to the well know commercial cloud solutions. If we would be successful with NextCloud and being able to make money based on that, we would most definitely consider to become an official supporter of NextCloud. This would be just a normal thing to do.

But if I decide to promote NextCloud to our customers, it will be most likely just a matter of days until someone bumps into this (or some similar) issue. Do you really expect me to explain my customers than: "Yeah, sorry. This is an open bug in Nextcloud for almost a decade now. For some reason, they have tied this simple and basic functionality #1175 to some niche problem where they are waiting for Apple to fix some of their implementations, which of course they never will (because why should Apple care?), so the Nextcloud-guys will never fix this problem, either. Better you just get used to it!"

If I do this, I can close my company just right there. So unfortunately my test with NextCloud seems to end here with the result: Promoting this product will most likely backfire and undermine our customers trust in us, so we simply can't!

**Comment by SebastianKrupinski** [comment]
Hi Everyone,

Would someone be so kind as to create a group with the "MacOS Contact app" and post the vCard for us.

Thank you

**Comment by patzm** [comment]
I just tried on macOS 15.0.1 (24A348) without success. I can see the option in the context menu but when I click it, no response.
![Image](https://github.com/user-attachments/assets/aa0075ea-a6c5-4818-ac36-948b4a606560)

the group also doesn't appear on NC and the NC groups are not shown in the Contacts app.

**Comment by patzm** [comment]
actually creating a group on iOS works. It is called _lists_ there. However also that _list_ doesn't appear as a property / group / field in the NextCloud web contacts viewer.
[Test.vcf.zip](https://github.com/user-attachments/files/17381655/Test.vcf.zip)

**Comment by SebastianKrupinski** [comment]
Thank you @patzm 

Looks like MacOS is using the multiple cards in a single vcf file as a grouping mechanism and NOT the KIND="group" property with references. (face palm)

```
BEGIN:VCARD
    VERSION:3.0
    PRODID:-//Apple Inc.//iPhone OS 18.0.1//EN
    N:Contact;Empty;;;
    FN:Empty Contact
    ORG:Nextcloud Is Awesome ;
END:VCARD
BEGIN:VCARD
    VERSION:3.0
    PRODID:-//Apple Inc.//iPhone OS 18.0.1//EN
    N:;Another;;;
    FN:Another
END:VCARD
```

**Comment by favincen** [comment]
> Looks like MacOS is using the multiple cards in a single vcf file as a grouping mechanism and NOT the KIND="group" property with references. (face palm)

I have the same output here (macOS 14.7), but I also suspect that macOS Contacts app list export feature is in fact exporting members instead of exporting the list/group itself...
And on my side too groups created within Apple Contacts app are not showing on NC side. So I see no way to export an apple made group vcard appart from actually sniffing a CardDAV transaction between the two...  😢

**Comment by patzm** [comment]
yes agreed, with @favincen: the _exported_ file might not actually be the group but just the chained contacts therein. Reason: I can't click export if the group is empty. It doesn't show the option.

**Comment by SebastianKrupinski** [comment]
@favincen @patzm do you have access to your NC database? To see if the entry gets created? It should be the last entry in the 'oc_cards' table when you create a new list. The 'carddata' field has the contents

**Comment by patzm** [comment]
My table is actually just called `cards` without the `oc_` prefix.
the table has 1036 entires, sounds about right.

What I did:
* create a _list_ called `test-contacts-ios` on my iPhone
* create a _group_ called `test-contacts-macos` on my Mac

Run 
```sql
SELECT * FROM cards WHERE carddata LIKE '%test-contacts%'
```

Result:


| id   | addressbookid | carddata                                                                                                                                                                                                | uri                                      | lastmodified | etag                             | size | uid                                  |
|------|---------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|--------------|----------------------------------|------|--------------------------------------|
| 1301 | 2             | BEGIN:VCARD VERSION:3.0 X-ADDRESSBOOKSERVER-KIND:group PRODID:-//Apple Inc.//iOS 18.0.1//EN N:test-contacts-ios;;;; FN:test-contacts-ios UID:0E85507C-C67C-4C12-B4C2-E5EF19CF4E37 END:VCARD              | B887B012-9C30-42E2-AE56-E6719234BDCD.vcf | 1729020074   | c6c7ed53c63b9f19f9e055194868d080 | 196  | 0E85507C-C67C-4C12-B4C2-E5EF19CF4E37 |

This table captures the row with `test-contacts` in the `carddata` column.

**Comment by SebastianKrupinski** [comment]
@patzm Thanks a million! This is great news, looks like Apple is using the proper vCard KIND property over CardDAV and our back end already supports it. Now just a matter of making a whole lot of changes to the front end.

**Comment by patzm** [comment]
Well the least anyone can do that uses your great open source software ❤️.

Topic wise: don't you worry about the inconsistentcy? iOS 📱 uploaded something, macOS didn't. 😕

**Comment by SebastianKrupinski** [comment]
> Topic wise: don't you worry about the inconsistentcy? iOS 📱 uploaded something, macOS didn't. 😕

Yes, this could be an issue, but it could just be a mater of us providing a proper CalDAV capability, before iOS will do proper groups, or it could just be that iOS does not support groups, this one of many clients that need to be tested, but this is a start.

**Comment by ilpssun** [comment]
Probably related: #3820

**Comment by xeniacloud** [comment]
Confirmed still reproducing on Contacts app 8.7.6 / Nextcloud 34.0.3, self-hosted instance backed by an LDAP/lldap directory. Groups created in the Contacts web UI are stored via CATEGORIES only, so they don't appear as Groups in macOS/iOS Contacts. Would love to see frontend support for writing/reading X-ADDRESSBOOKSERVER-KIND:group vCards land, per the backend support mentioned above.
